frontend/src/features/drug_discovery/components/ligands/LigandsList.vue

```diff
@@ -0,0 +1,281 @@
+          const file = this.uploadingLigandFiles[index];
+          const metaData = additionalMetaDataArray ? additionalMetaDataArray[index] : undefined;
+          const newLigand = await this.drugDiscoveryStore.uploadLigandToExperiment(this.experimentId, file, metaData);
+          this.ligands.push(newLigand!);
+        }
+        this.uploadLigandDialog = false;
+        this.uploadingLigandFiles = [];
+      },
+      async getLigandData(ligand: ExtendedLigandMetaData) {
+        ligand.loadingLigandData = true;
+        try {
+          ligand.data = await this.drugDiscoveryStore.fetchLigandDataForExperiment(this.experimentId, ligand.ligand_id);
+        } catch (error) {
+          console.error('Error fetching ligand data:', error);
+        } finally {
+          ligand.loadingLigandData = false;
+          this.uploadingLigandFiles = [];
+        }
+      },
+      async deleteLigand(ligandToDelete: ExtendedLigandMetaData) {
+        this.$q.loading.show({
+          spinner: QSpinnerOrbit,
+          message: `Deleting ligand`
+        });
+        try {
+          await this.drugDiscoveryStore.deleteLigandFromExperiment(this.experimentId, ligandToDelete.ligand_id);
+          this.ligands = this.ligands.filter(ligand => ligand.ligand_id !== ligandToDelete.ligand_id);
+        } catch (error) {
+          console.error('Error deleting ligand:', error);
+        } finally {
+          this.$q.loading.hide();
+        }
+      },
+      toggleSelectAll(value: boolean) {
+        if (value) {
+          // Replace the array with a new one containing all IDs to ensure reactivity
+          this.selectedTargets = this.drugDiscoveryStore.targets.map(target => target.target_id);
+        } else {
+          // Clear the array by replacing it with a new empty array
+          this.selectedTargets = [];
+        }
+      },
+
+      async openTargetSelectionDialog(ligand: ExtendedLigandMetaData) {
+        this.selectedLigand = ligand;
+        this.targetSelectionDialogVisible = true;
+        this.selectedTargets = [];
+        this.selectAllTargets = false;
+        this.targets = await this.drugDiscoveryStore.fetchTargetsForExperiment(this.experimentId) as ExtendedTargetMetaData[];
+      },
+
+      async addLigandsToSelectedTargets() {
+        const store = useDrugDiscoveryStore();
+
+        for (let target_id of this.selectedTargets) {
+          if (this.selectedLigand && this.selectedLigand.data) {
+            // Assuming `selectedLigand.data` has the required SDF file or you have it available somehow
+            const sdfFileContent = this.selectedLigand.data.sdf_file;
+
+            const file = new File([new Blob([sdfFileContent!])], this.selectedLigand.ligand_name + ".sdf")
+
+            try {
+              await store.uploadLigandToTarget(this.experimentId, target_id.toString(), file);
+              // Handle success - Maybe show a notification?
```

Comment:
```suggestion
              // TODO Show notification
```

TODOs can be scanned by Intellij and vscode IDEs and show you suggestions